src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -139,7 +140,7 @@ public void OnResponseHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
                         }
 
                         if (value.Length != 3)
-                            throw new Exception("Invalid status code");
+                            throw new HttpRequestException(SR.net_http_invalid_response);
```

Comment:
Is there a way to have an inner exception to capture this string, "Invalid status code"? Every time we see the `SR.net_http_invalid_response error`, we always ask ourselves, "what was invalid in the response?".  I feel like this is important diagnostic information we will lose here. At a minimum maybe we need to use NetEventSource.TraceError or something. But I would also want to capture this information in an inner exception.

Reply:
I think it'd be reasonable to do a pass through all of our SocketHttpHandler exceptions and add more specific error messages about what the issue is.  This is just one occurrence of it.

Reply:
Can you leave the "Invalid status code" text in the code so that we can find it later?, i.e.

```c#
throw new HttpRequestException(SR.net_http_invalid_response); // "Invalid status code"
```

Reply:
Sure.